lib/lumberjack/beats/server.rb

```diff
@@ -375,17 +380,29 @@ def close
     end
 
     def data(map, &block)
-      block.call(map) if block_given?
+      block.call(map, identity_stream(map)) if block_given?
     end
 
     def reset_next_ack(window_size)
-      klass = (@version == Parser::PROTOCOL_VERSION_1) ? AckingProtocolV1 : AckingProtocolV2
+      klass = version_1? ? AckingProtocolV1 : AckingProtocolV2
       @ack_handler = klass.new(window_size)
     end
 
     def send_ack(sequence)
       @fd.syswrite(@ack_handler.ack_frame(sequence))
     end
+
+    def version_1?
+      @version == Parser::PROTOCOL_VERSION_1
+    end
+
+    def identity_stream(map)
+      [peer,
+       map.fetch("beats", {})["name"],
+       map["type"],
+       map["input_type"],
+       map["source"]].compact.join("-")
```

Comment:
not sure about these fields for stream identity.
See fields documentation for [packetbeat](https://github.com/elastic/packetbeat/blob/master/etc/fields.yml), [filebeat](https://github.com/elastic/filebeat/blob/master/etc/fields.yml) and [topbeat](https://github.com/elastic/filebeat/blob/master/etc/fields.yml) .

Only common fields you will get from all beats:
  @metadata.beats,
  @metadata.type   (which is similar to map["type"]
  beat.name

These fields are not enough to build a unique ID for filebeat based logs though.
filebeat specific:
   input_type  (not sure if really required)
   source

Not sure if stream identifier can be of interest for packetbeat based data. Topbeat likely not, as topbeat is about some metrics.

[winlogbeat](github.com/elastic/winlogbeat) might export other [fields](https://github.com/elastic/winlogbeat/blob/master/eventlog/eventlog.go#L92). No idea if multiline will be of interest to windows event logs. Due to winlogbeat potentially collecting logs from remote hosts, you must also include:
  - computerName
  - sourceName

@andrewkroh might have some more input on unique stream identifier for winlogbeat.